nav2_rviz_plugins/src/nav2_panel.cpp

```diff
@@ -381,6 +567,189 @@ Nav2Panel::~Nav2Panel()
+    store_initial_pose_ = false;
+  }
+}
+
+bool Nav2Panel::isLoopValueValid(std::string & loop_value)
+{
+  // Check for just empty space
+  if (loop_value.empty()) {
+    std::cout << "Loop value cannot be set to empty, setting to 0" << std::endl;
+    loop_value = "0";
+    nr_of_loops_->setText("0");
+    return true;
+  }
+
+  // Check for any chars or spaces in the string
+  for (char & c : loop_value) {
+    if (isalpha(c) || isspace(c) || ispunct(c)) {
+      waypoint_status_indicator_->setText(
+        "<b> Note: </b> Set a valid value for the loop");
+      std::cout << "Set a valid value for the loop, check for alphabets and spaces" << std::endl;
+      navigation_mode_button_->setEnabled(false);
+      return false;
+    }
+  }
+
+  try {
+    stoi(loop_value);
+  } catch (std::invalid_argument const & ex) {
+    // Handling special symbols
+    waypoint_status_indicator_->setText("<b> Note: </b> Set a valid value for the loop");
+    navigation_mode_button_->setEnabled(false);
+    return false;
+  } catch (std::out_of_range const & ex) {
+    // Handling out of range values
+    waypoint_status_indicator_->setText(
+      "<b> Note: </b> Loop value out of range, setting max possible value"
+    );
+    loop_value = std::to_string(std::numeric_limits<int>::max());
+    nr_of_loops_->setText(QString::fromStdString(loop_value));
+  }
+  return true;
+}
+
+void Nav2Panel::loophandler()
+{
+  loop_no_ = nr_of_loops_->displayText().toStdString();
+
+  // Sanity check for the loop value
+  if (!isLoopValueValid(loop_no_)) {
+    return;
+  }
+
+  // Enabling the start_waypoint_follow button, if disabled.
+  navigation_mode_button_->setEnabled(true);
+
+  // Disabling nav_through_poses button
+  if (!loop_no_.empty() && stoi(loop_no_) > 0) {
+    pause_resume_button_->setEnabled(false);
+  } else {
+    pause_resume_button_->setEnabled(true);
+  }
+}
+
+void Nav2Panel::handleGoalLoader()
+{
+  acummulated_poses_.clear();
+
+  std::cout << "Loading Waypoints!" << std::endl;
+
+  QString file = QFileDialog::getOpenFileName(
+    this,
+    tr("Open File"), "",
+    tr("yaml(*.yaml);;All Files (*)"));
+
+  YAML::Node available_waypoints;
+
+  try {
+    available_waypoints = YAML::LoadFile(file.toStdString());
+  } catch (const std::exception & ex) {
+    std::cout << ex.what() << ", please select a valid file" << std::endl;
+    updateWpNavigationMarkers();
+    return;
+  }
+
+  const YAML::Node & waypoint_iter = available_waypoints["waypoints"];
+  for (YAML::const_iterator it = waypoint_iter.begin(); it != waypoint_iter.end(); ++it) {
+    auto waypoint = waypoint_iter[it->first.as<std::string>()];
+    auto pose = waypoint["pose"].as<std::vector<double>>();
+    auto orientation = waypoint["orientation"].as<std::vector<double>>();
+    acummulated_poses_.push_back(convert_to_msg(pose, orientation));
+  }
+
+  // Check for goal poses, if they are not in an obstacle
+  for (unsigned int i = 0; i < acummulated_poses_.size(); ++i) {
```

Comment:
Remove this - lets just make this application "dumb"

Reply:
+ remove the collision checker / costmap / footprint / etc subscribers.

Do we need TF?

Reply:
Sure, we can scale it down. The reason I assume that you didn't want to have it is maybe because collision_checker might make the plugin a bit heavy ?

Reply:
Yeah as well as the case that it may be likely pre-defined paths could have blocked waypoints, but I think that's for the WPF to handle, not the visualization / command-control application's role